src/actions/heap/heap.ts

```diff
@@ -82,47 +94,62 @@ export class HeapAction extends Hub.Action {
       .property_type as HeapPropertyType
     const envId = request.formParams.env_id!
     const heapFieldLabel: string = request.formParams.heap_field!
-    let identityField: Hub.Field
+    const webhookId = request.webhookId !== undefined ? request.webhookId : "unknown"
+    const logTag: LogTag = {
+      envId,
+      webhookId,
+    }
+    let identityField: Hub.Field | undefined
 
     let fieldMap: LookerFieldMap = {} as LookerFieldMap
-    const heapField = this.resolveHeapField(propertyType)
-    const requestUrl = this.resolveApiEndpoint(propertyType)
+    const heapField = this.resolveHeapField(propertyType, logTag)
+    const requestUrl = this.resolveApiEndpoint(propertyType, logTag)
     const errors: Error[] = []
+    const requestPromises: Promise<void>[] = []
     let rowsProcessed = 0
     let rowsReceived = 0
     let requestBatch: HeapEntity[] = []
-    const requestPromises: Promise<void>[] = []
 
     await request.streamJsonDetail({
       onFields: (fieldset) => {
-        winston.info(`envId ${envId} fieldset ${JSON.stringify(fieldset)}`)
-        const allFields = Hub.allFields(fieldset)
-        winston.info(`envId ${envId} allFields ${JSON.stringify(allFields)}`)
-        identityField = this.extractHeapFieldByLabel(allFields, heapFieldLabel)
-        winston.info(`envId ${envId} heapFieldName ${identityField} heapFieldLabel ${heapFieldLabel}`)
-        fieldMap = this.extractFieldMap(allFields)
-        winston.info(`envId ${envId} fieldMap ${JSON.stringify(fieldMap)}`)
+        try {
+          winston.debug(`envId ${envId} fieldset ${JSON.stringify(fieldset)}`, logTag)
+          const allFields = Hub.allFields(fieldset)
+          winston.debug(`envId ${envId} allFields ${JSON.stringify(allFields)}`, logTag)
+          identityField = this.extractHeapFieldByLabel(allFields, heapFieldLabel)
+          winston.debug(`envId ${envId} heapFieldName ${identityField} heapFieldLabel ${heapFieldLabel}`, logTag)
+          fieldMap = this.extractFieldMap(allFields)
+          winston.debug(`envId ${envId} fieldMap ${JSON.stringify(fieldMap)}`, logTag)
+        } catch(err) {
+          errors.push(new Error('Encountered errors in processing fields: ' + getErrorMessage(err)))
+        }
       },
       onRow: (row) => {
+        if (!identityField) {
```

Comment:
why are we silently dropping rows if the identity field is not set?

Reply:
`identityField` is a required field for each row. If it's empty, heapFieldValue is empty, it will quit either way. choosing to quit early because `extractRequiredField` will throw an error if the value is empty.